### PR TITLE
Update lazy-definitions.md

### PR DESCRIPTION
**Pull request by KPCOFGS**
Fixed a typo in the link and now it will redirect users to the expected place

### CODE REVIEW

File: docs/lazy-definitions.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ If you have a set of mutually recursive functions, you only need to make one of
 
 ### Automatic optimization
 
-HVM-lang carries out [match linearization](compiler-options#linearize-matches) and [combinator floating](compiler-options#float-combinators) optimizations, enabled through the CLI, which are active by default in strict mode.
+HVM-lang carries out [match linearization](compiler-options.md#linearize-matches) and [combinator floating](compiler-options#float-combinators) optimizations, enabled through the CLI, which are active by default in strict mode.
```
Comment (developedby) [inline docs/lazy-definitions.md:62]:
```suggestion
HVM-lang carries out [match linearization](compiler-options.md#linearize-matches) and [combinator floating](compiler-options.md#float-combinators) optimizations, enabled through the CLI, which are active by default in strict mode.
```
I see one more spot actually

**Review by developedby — CHANGES_REQUESTED** [review]

**Comment by KPCOFGS** [comment]
Thank you for your feedback! I just fixed these typos as well

**Review by developedby — APPROVED** [review]
Thank you for your contribution!

**Comment by KPCOFGS** [comment]
Edit: Another link typo

**Comment by KPCOFGS** [comment]
Sorry about some sudden changes. I just fixed another link and I noticed there is a missing readme file `automatic-vectorization-with-tagged-lambdas`

**Comment by developedby** [comment]
No problem, thank you for your time.
The missing link is for a feature we used to have that is not supported at the moment.
I removed the document but didn't see that it was referenced here.

**Pull request merged**